### ISSUE TITLE
httpx adding extra proxy headers?

### ISSUE THREAD

**Issue by GnubiBORED** [encode/httpx#474]
`  proxy = httpx.HTTPProxy(
                    #la variable ip que pasamos a la funcion es el proxy
                    "http://{}".format(ip),
                    #y aqui añadimos los headers especificos que llevan las peticiones CONNECT cuanndo usamos proxy
                    proxy_headers={"Host": url+':443',
                                   "Proxy-Connection":"Keep-Alive"}`

im sniffing these with charlesproxy and output is

`CONNECT www.host.es:443 HTTP/1.1
host: https://www.host.es/m/m/login?:443
user-agent: python-httpx/0.7.4
accept-encoding: gzip, deflate, br
connection: keep-alive
proxy-connection: Keep-Alive
accept: */*

`
i would like to remove all unless proxy-connection and host

**Comment by florimondmanca** [comment]
Yes, HTTPX adds various headers automatically when making requests. I don't know if `CONNECT` should be an exception. @sethmlarson What's your opinion on this?

**Comment by florimondmanca** [comment]
@GnubiBORED I updated the title of this issue towards what I understood the underlying issue was. Feel free to correct me if that's not what you meant.

**Comment by GnubiBORED** [comment]
yes its exactly what i meant, and in my opinion and in my case use, not just in connect, if in all methods every header is manipulable, better, i mean if i can customize the request more, better, in these case FOR ME its preferible to remove those extra headers i dont need .

**Comment by florimondmanca** [comment]
Do you just *not need them* (in which case we can probably just leave them in?), or do they pose an issue with your use case?

**Comment by GnubiBORED** [comment]
> 
> 
> Do you just _not need them_ (in which case we can probably just leave them in?), or do they pose an issue with your use case?

its an issue for my case cause im sniffing and copying http request and in the original ones there are just 2 headers so i need to remove the rest, if not then i cannot do the requests in same way and its a problem for me

**Comment by tomchristie** [comment]
> btw i didnt check all possible cases for post and get, but if it adds headers in any case, its a problem for me too

We absolutely need to include some default headers, eg. content-length, a default user-agent, etc... however we *also* include lower level APIs for building and sending request instances explicitly.

**Comment by GnubiBORED** [comment]
> 
> 
> > btw i didnt check all possible cases for post and get, but if it adds headers in any case, its a problem for me too
> 
> We absolutely need to include some default headers, eg. content-length, a default user-agent, etc... however we _also_ include lower level APIs for building and sending request instances explicitly.

adding default headers isnt a problem but isnt possible to remove them if needed?

**Comment by tomchristie** [comment]
I'm not sure *what* additional headers we ought to be including on "CONNECT" proxy requests?
It may be that we ought to be supressing the default request headers in that case.
Are `accept-encoding`, `accept`, and `connection` meaningful in that context?
Is `user-agent` typically included by clients on "CONNECT" requests?

It might help to know what the behavior of `curl` and `requests` is in equivelent cases here?

**Comment by tomchristie** [comment]
The most sensible way for us to resolve this issue would be to double check if we've got the same behavior as `requests`/`urllib3` here or not.

@GnubiBORED would you be up for taking a look into that?

**Comment by yeraydiazdiaz** [comment]
Moving this issue to httpcore.

I've checked behaviour with urllib3 which relies on the stdlib's `http` module which [does _not_ include the request headers in the CONNECT request](https://github.com/python/cpython/blob/3.8/Lib/http/client.py#L884-L893) but allows the caller to [set ones explicitly](https://docs.python.org/3/library/http.client.html#http.client.HTTPConnection.set_tunnel).
